fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -204,20 +204,37 @@ public DatatableData retrieveDatatable(final String datatable) {
     }
 
     @Override
-    public List<JsonObject> queryDataTable(String datatable, String columnFilter, String valueFilter, String resultColumns) {
-        Arrays.asList(datatable, columnFilter, valueFilter, resultColumns).forEach(SQLInjectionValidator::validateDynamicQuery);
+    public List<JsonObject> queryDataTable(String datatable, String columnFilter, String valueFilter, String resultColumns, Integer offset,
+            Integer limit, String orderBy, String sortOrder) {
+        Arrays.asList(datatable, columnFilter, valueFilter, resultColumns, orderBy, sortOrder)
+                .forEach(SQLInjectionValidator::validateDynamicQuery);
 
         List<ResultsetColumnHeaderData> resultsetColumnHeaderData = genericDataService.fillResultsetColumnHeaders(datatable);
-        validateRequestParams(columnFilter, valueFilter, resultColumns, resultsetColumnHeaderData);
+        validateRequestParams(columnFilter, valueFilter, resultColumns, resultsetColumnHeaderData, orderBy, sortOrder);
+
+        StringBuilder sql = new StringBuilder("select " + resultColumns + " from " + datatable + " where " + columnFilter + " = ?");
```

Comment:
String concatenation is inherently insecure (read: very easy to inject other SQL statements). Please provide a more secure way to assemble the query.

Reply:
thanks for review working on it.

Reply:
well, I am already sanitizing the inputs, it might save?
tried to implement other more secured ways but getting stucked in passing paramList as calling other method eg: callFilteredPgSql